engine/resource/src/resource_chunk_cache.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DM_RESOURCE_CHUNK_CACHE_H
+#define DM_RESOURCE_CHUNK_CACHE_H
+
+#include <stdint.h>
+#include <dlib/hash.h> // dmhash_t
+
+struct ResourceCacheChunk
+{
+    uint8_t* m_Data;
+    uint32_t m_Offset;
+    uint32_t m_Size;
+};
+
+typedef struct ResourceChunkCache* HResourceChunkCache;
+
+HResourceChunkCache ResourceChunkCacheCreate(uint32_t max_memory);
+void                ResourceChunkCacheDestroy(HResourceChunkCache cache);
```

Comment:
Still an internal api, but I opted for a C api to start with, as it's easier to port to C++ using our code wrapper generator.